### ISSUE TITLE
Updating remaining block count during wallet sync isn't continuous

### ISSUE THREAD

**Issue by akuukka** [monero-project/monero-gui#1552]
Not sure if daemon sync has the same problem (I always run the daemon), but when I open the GUI and it has to sync my wallet, the remaining block count behaves strangely.

It quickly goes down by several thousand block, then stops completely for a long time. Then again it updates itself later and goes down by several thousands. Rinse and repeat.

It's really frustrating. Isn't it possible to update the remaining block count more "continuously"?

I'm running Monero on external HDD (yeah, initial sync was really painful but I can now easily keep up with the chain) so the wallet sync is quite slow for me. It would be super cool and user friendly to know the progress better.

**Comment by skftn** [comment]
GUI version, OS?

**Comment by akuukka** [comment]
0.12.3.0 OS X

Reproducing might require an external HDD and wallet which is behind at least 10000 blocks.

**Comment by akuukka** [comment]
USB-C on a brand new MacBook Pro

**Comment by skftn** [comment]
How long does the total wallet refresh take, do you reckon?

**Comment by skftn** [comment]
I'm going to assume `monerod` is busy syncing the blockchain whilst the GUI is trying to refresh the wallet.

Try the following:

1. Open GUI->Settings
2. Press 'Stop Local Node' (if it is started)
3. Navigate to 'Blockchain location' -> Show advanced
4. Set input value to `--max-concurrency 2`
5. Press 'Start Local Node'

`monerod` should now take up less resources (I/O, CPU) and I suspect this would make your wallet refresh faster.

**Comment by akuukka** [comment]
Yes, we are talking about wallet sync now. My blockhain is always synced because I run monerod 24/7.

But if I haven't opened the wallet GUI in about a week, it takes about 1-2 minutes to sync the wallet and as I mentioned, the progress indicator tends to stop at times.

I don't think this is about concurrency, because my older dual core MacBook Pro behaves almost exactly the same as my newer quad core MacBook Pro. It's clear that the disk I/O is the bottleneck because the newer computer is only slightly faster.

**Comment by lacksfish** [comment]
I see the same UI/UX issue. Unlocked balance doesn't update while syncing a few thousand blocks. This is on 0.13.0.4 Unlocked balance should jump back to full balance after only syncing the next 10 blocks after spending.

**Comment by dEBRUYNE-1** [comment]
Can you check whether this is still an issue with GUI v0.14.1.0?

**Comment by lacksfish** [comment]
Will do

**Comment by dEBRUYNE-1** [comment]
Thanks.

**Comment by dEBRUYNE-1** [comment]
@lacksfish - ping.

**Comment by lacksfish** [comment]
Sorry sorry, will get to it.

**Comment by dEBRUYNE-1** [comment]
Thanks!

**Comment by lacksfish** [comment]
Giving it a spin now.

**Comment by lacksfish** [comment]
Syncing, got my eyes on it

**Comment by lacksfish** [comment]
Regarding balance during sync:

So it seems to refresh while syncing, but I have a bunch of tx's where I churned back into my own Ledger wallet, and the sync seems to always *add* the amount I churned, but is not subtracting the amount sent out correctly? Otherwise, I can not explain it, but the balance during sync is too high.

Block count is moving along just fine, no big random jumps, very consistent.
